### PR TITLE
Feat: Support Gorilla APIBench 

### PR DESCRIPTION
**Pull request by yueqis**
I added support for evaluation of Gorilla's APIBench.
Gorilla's APIBench source repo: `https://github.com/ShishirPatil/gorilla/tree/main`.
Gorilla's APIBench arxiv paper: https://arxiv.org/pdf/2305.15334.
All code for evaluation is contained in the directory evaluation/gorilla.
Gorilla's APIBench contains three subsets: testing apis from huggingface, torch, and tensorflow. So for the hubs argument below, one could choose from either hf for huggingface, th for torch, and tf for tensorflow.
To run evaluation, start from the root directory, and run `bash evaluation/gorilla/scripts/run_infer.sh $llm_config $agent $eval_limit $hubs`, for example `bash evaluation/gorilla/scripts/run_infer.sh llm CodeActAgent 1 hf`. See more details in `run_infer.sh`.
I kept the copyright headers in the PR, as this the gorilla repo require keeping the license when using their code.
Removed all unnecessary files from the PR.

### CODE REVIEW

**Review by yufansong — COMMENTED** [review]
Most code LGTM expect we need to totally copy the code from other repo. But I also don't have better solution to solve it. Hope some other guys also can have a look on it.

File: evaluation/gorilla/ast_eval_hf.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Copyright 2023 https://github.com/ShishirPatil/gorilla
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (li-boxuan) [inline evaluation/gorilla/ast_eval_hf.py:3]: According to Apache License v2.0 (https://www.apache.org/licenses/LICENSE-2.0):

> You must cause any modified files to carry prominent notices stating that You changed the files.

If I understand correctly, you have made some modifications to their source code. Could you please state it in the header?

**Review by li-boxuan — COMMENTED** [review]
Hi, could you please include a simple README of how to run this benchmark (take a look at others, especially SWE-Bench's), and how to obtain benchmark reports/results?

**Review by neubig — COMMENTED** [review]
Hi @yueqis , thanks for the contribution! This looks good to me, but I agree we should add a `README.md` file to explain what the benchmark is and how to run it, patterned after the one for swe_bench. Once that's done I think this should be ready to merge.

File: evaluation/gorilla/run_infer.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        'correct': correct, 
+        'hallucination': hallucination, 
+        'answer_id': 'None', 
+        'model_id': metadata['model_name'], 
+        'metadata': metadata,
+        'history': [
+            (event_to_dict(action), event_to_dict(obs)) for action, obs in state.history
+        ],
+        'error': state.error if state and state.error else None,}
+    config.workspace_mount_path = old_workspace_mount_path
```
Comment (li-boxuan) [inline evaluation/gorilla/run_infer.py:148]: Please apply a similar fix here: https://github.com/OpenDevin/OpenDevin/pull/2214

File: evaluation/gorilla/ast_eval_hf.py (hunk 1 above)
Comment (yueqis) [reply evaluation/gorilla/ast_eval_hf.py:3]: Added, thanks so much!

File: evaluation/gorilla/run_infer.py (hunk 2 above)
Comment (yueqis) [reply evaluation/gorilla/run_infer.py]: Added, thanks so much!

**Comment by tobitege** [comment]
@yueqis in your local repo, run "make lint" ;)

**Comment by yufansong** [comment]
Add some code for lint problem. Good to merged now.

**Pull request merged**